### ISSUE TITLE
SA1010 false positive when list syntax is used in a field initializer

### ISSUE THREAD

**Issue by AArnott** [DotNetAnalyzers/StyleCopAnalyzers#3687]
The following line produces a SA1010 warning claiming that the open square bracket should not be preceded by a space:

```cs
    protected static readonly string[] DefaultMetadataPaths = [MetadataPath, WdkMetadataPath];
```

This is using 1.2.0.507.

**Comment by MiYanni** [comment]
I hit this when telling VS to make fixes for [Use collection initializers (IDE0028)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0028). It will automatically create statements that trigger SA1010. For example:
```C#
List<string> foo = [];
```

**Comment by dciborow** [comment]
Hit this in a few places.

```csharp
ctors = [.. ctors.OrderByDescending(info => info.GetParameters().Length)];
```

```csharp
object result = TypesManager.InvokeGenericStaticMethod(typeof(ServiceLocator), nameof(ResolveMock), [type]);
```

```csharp
if (request.OtherEmails != null)
{
    emails = [.. request.OtherEmails.Split([';', ','])];
}
```

```csharp
        public static readonly string[] AllBlastRadiusStrings = [SingleNode, SingleCustomer, SingleScaleUnitSlicePartition, SingleScaleUnit, SingleRack, SingleCluster, SingleAZ, SingleRegion, MultipleRacks, MultipleClusters, MultipleAZs, MultipleRegions, Global, Other];
```

**Comment by bjornhellander** [comment]
I assume that .NET 8 and C# 12 will be released very soon now. This fix is probably trivial, but what do you think about adding a new test project, @sharwell? Should it wait until you are done with your updates to the test projects (https://github.com/DotNetAnalyzers/StyleCopAnalyzers/pull/3675#issuecomment-1740054572), or can it be done now? If you would like to wait, do you have any estimate of when you will be done?

**Comment by DavidArno** [comment]
.NET 8 and C# 12 were released yesterday and so this is now an issue with production code. For now, I've had to disable the SA1010 rule for our code until this can be fixed. Please can this be fixed as quickly as possible.

**Comment by commonsensesoftware** [comment]
I'm unsure if all the various cases are being tracked or accounted for, but I thought I would report that SA1010 also fires during coalesce, coalesce assign, and ternary operations.

```c#
List<string>? list = default;

// fires on coalesce
List<string> otherList = list ?? [];

// fires on ternary
List<string> anotherList = list is null ? [] : list;

// fires on coalesce assigement
list ??= [];
```

**Comment by YoshiRulz** [comment]
see also #3503
